packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js

```diff
@@ -48,7 +51,16 @@ export function SidebarNavigationItemGlobalStyles( props ) {
 			{ ...props }
 			onClick={ () => {
 				// Switch to edit mode.
-				setCanvasMode( 'edit' );
+				history.push(
+					{
+						...params,
+						canvasMode: 'edit',
```

Comment:
Do the links for [patterns](https://github.com/WordPress/gutenberg/blob/4a625e5f097056d5f23c1a68385b3327573727bf/packages/edit-site/src/components/page-patterns/fields.js#L63) and [templates](https://github.com/WordPress/gutenberg/blob/4a625e5f097056d5f23c1a68385b3327573727bf/packages/edit-site/src/components/page-templates/fields.js#L38) also need `canvasMode: 'edit',`?

E.g.,

```js
	const { onClick } = useLink( {
		postType: item.type,
		postId: isUserPattern || isTemplatePart ? item.id : item.name,
		canvas: 'edit',
		canvasMode: 'edit',
	} );
```

Otherwise, clicking on individual patterns and templates lead to the list + preview mix, similar to the screenshot above.